src/main/java/frc/robot/commands/intake/CurrentBasedIntakeFeeder.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.commands.intake;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Feeder;
+import frc.robot.subsystems.IntakeSubsystem;
+
+public class CurrentBasedIntakeFeeder extends Command {
+    private final Feeder feeder;
+    private final IntakeSubsystem intake;
+    private final Timer timer = new Timer();
+    private boolean slowState = false;
+    private double intakeSpikeCount = 0;
+
+    public CurrentBasedIntakeFeeder(IntakeSubsystem intake, Feeder feeder) {
+        this.intake = intake;
+        this.feeder = feeder;
+        addRequirements(intake, feeder);
+    }
+
+    @Override
+    public void initialize() {
+        feeder.switchFeederBeamState(true);
+        timer.reset();
+        timer.start();
+    }
+
+    @Override
+    public void execute() {
+        intake.setMotorSpeed(Constants.INTAKE_MOTOR_1_SPEED, Constants.INTAKE_MOTOR_2_SPEED);
+        if(intake.getMotor1StatorCurrent() > Constants.INTAKE_PIECE_THESHOLD){
+            intakeSpikeCount++;
+        }
+        slowState = (intakeSpikeCount > Constants.INTAKE_SPIKE_THESHOLD);
+        double speed = slowState ? Constants.FEEDER_SLOW_SPEED : Constants.FEEDER_MOTOR_ENTER_SPEED;
+        feeder.setFeederMotorSpeed(speed);
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        intake.stopMotors();
+        feeder.stopFeederMotor();
+        feeder.switchFeederBeamState(false);
```

Comment:
I think it is logically wrong to do this here. This should be done whenever we're shooting.
Just as an example, with current code, you can't shoot the first game piece in auto.

Reply:
put in feederUntilGame...